play-json/jvm/src/test/scala/play/api/libs/json/JsonSpec.scala

```diff
@@ -60,66 +83,369 @@ class JsonSpec extends org.specs2.mutable.Specification {
+        val expectedPost = Post("foobar", Some(postDate))
+
+        Json.parse(postJson).as[Post](LenientPostFormat).
+          aka("parsed") must_== expectedPost
+      }
+
+      "with default/lenient date format with no millis and ISO8601 zone" in {
+        val cal = Calendar.getInstance(TimeZone getTimeZone "UTC")
+        cal.setTime(postDate)
+        cal.add(Calendar.HOUR_OF_DAY, -7)
+
+        val postJson =
+          """{"body": "foobar", "created_at": "2011-04-22T13:33:48+0700"}"""
+        val expectedPost = Post("foobar", Some(cal.getTime))
+
+        Json.parse(postJson).as[Post](LenientPostFormat).
+          aka("parsed") must_== expectedPost
+      }
+
+      "with default/lenient date format with millis" in {
+        val postJson =
+          """{"body": "foobar", "created_at": "2011-04-22T13:33:48.000"}"""
+        val expectedPost = Post("foobar", Some(postDateWithTZ(TimeZone.getDefault)))
+
+        Json.parse(postJson).as[Post](LenientPostFormat).
+          aka("parsed") must_== expectedPost
+      }
+
+      "with default/lenient date format without millis or time zone" in {
+        val postJson =
+          """{"body": "foobar", "created_at": "2011-04-22T13:33:48"}"""
+        val expectedPost = Post("foobar", Some(postDateWithTZ(TimeZone.getDefault)))
+
+        Json.parse(postJson).as[Post](LenientPostFormat).
+          aka("parsed") must_== expectedPost
+      }
     }
 
-    "with default/lenient date format with no millis and UTC zone" in {
-      val postJson =
-        """{"body": "foobar", "created_at": "2011-04-22T13:33:48Z"}"""
-      val expectedPost = Post("foobar", Some(postDate))
-
-      Json.parse(postJson).as[Post](LenientPostFormat).
-        aka("parsed") must_== expectedPost
-    }
-
-    "with default/lenient date format with no millis and ISO8601 zone" in {
-      val cal = Calendar.getInstance(TimeZone getTimeZone "UTC")
-      cal.setTime(postDate)
-      cal.add(Calendar.HOUR_OF_DAY, -7)
-
-      val postJson =
-        """{"body": "foobar", "created_at": "2011-04-22T13:33:48+0700"}"""
-      val expectedPost = Post("foobar", Some(cal.getTime))
-
-      Json.parse(postJson).as[Post](LenientPostFormat).
-        aka("parsed") must_== expectedPost
-    }
-
-    "with default/lenient date format with millis" in {
-      val postJson =
-        """{"body": "foobar", "created_at": "2011-04-22T13:33:48.000"}"""
-      val expectedPost = Post("foobar", Some(postDateWithTZ(TimeZone.getDefault)))
-
-      Json.parse(postJson).as[Post](LenientPostFormat).
-        aka("parsed") must_== expectedPost
-    }
-
-    "with default/lenient date format without millis or time zone" in {
-      val postJson =
-        """{"body": "foobar", "created_at": "2011-04-22T13:33:48"}"""
-      val expectedPost = Post("foobar", Some(postDateWithTZ(TimeZone.getDefault)))
-
-      Json.parse(postJson).as[Post](LenientPostFormat).
-        aka("parsed") must_== expectedPost
+    "when parsing numbers" in {
```

Comment:
There are some isolated tests for parsing numbers but decided to add these here since the changes made have an effect and number types.

These tests are also closer to real usage of play-json.